extensions/http-curl/sitetosite/HTTPProtocol.cpp

```diff
@@ -0,0 +1,310 @@
+/**
```

Comment:
General organizational note --it seems to be a bit of a mixing of concerns to have s2s-specific code under the http-curl extension.

Granted, we probably have/will have dependencies between extensions which may or may not be easy to express using cmake constructs.

Reply:
this HTTPProtocol implementation uses CURL to do its work, hence why it exists here. While we could certainly have cross compilation between extensions, it doesn't seem like it's worth the increase in build time to change. One thing I want to avoid is having single function extensions and instead of have classes of extensions.